### PR TITLE
Show Bottomsheet when no notification permission

### PR DESCRIPTION
**Pull request by sujitacharya2005**
Resolves https://github.com/brave/brave-browser/issues/27795

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

https://user-images.githubusercontent.com/32419898/217641827-3165238b-2736-47f0-9998-568284721b27.mp4

### CODE REVIEW

File: android/java/res/drawable/you_are_not_earning_rewards_background.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) 2022 The Brave Authors. All rights reserved.
```
Comment (deeppandya) [inline android/java/res/drawable/you_are_not_earning_rewards_background.xml:2]: nit : year

Comment (sujitacharya2005) [reply android/java/res/drawable/you_are_not_earning_rewards_background.xml]: updated

File: android/java/res/layout/rewards_you_are_not_earning_dialog.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        app:layout_constraintTop_toBottomOf="@+id/notification_disabled_message" />
+
+    <ImageView
+        android:id="@+id/notification_disabled_close_button"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:clickable="true"
+        android:contentDescription="@null"
+        android:focusable="true"
+        android:src="@drawable/ic_baseline_close_24"
```
Comment (SergeyZhukovsky) [inline android/java/res/layout/rewards_you_are_not_earning_dialog.xml:57]: `srcCompat`

Comment (sujitacharya2005) [reply android/java/res/layout/rewards_you_are_not_earning_dialog.xml]: updated

**Review by SergeyZhukovsky — APPROVED** [review]
++

File: android/java/res/layout/brave_rewards_panel_layout.xml [hunk 3]
```diff
@@ -19,10 +19,13 @@
                     android:layout_width="match_parent"
                     android:layout_height="wrap_content">
 
+
                     <include
                         android:id="@+id/brave_rewards_panel_notification_layout_id"
                         layout="@layout/brave_rewards_panel_notification_layout" />
 
+
```
Comment (deeppandya) [inline android/java/res/layout/brave_rewards_panel_layout.xml:27]: nit : new line should be removed here

File: android/java/res/layout/brave_rewards_panel_notification_layout.xml [hunk 4]
```diff
@@ -48,6 +53,9 @@
                 app:srcCompat="@drawable/ic_close_12"/>
 
         </FrameLayout>
+
```
Comment (deeppandya) [inline android/java/res/layout/brave_rewards_panel_notification_layout.xml:56]: nit : we should remove extra new lines here.

File: android/java/res/layout/brave_rewards_panel_notification_layout.xml [hunk 5]
```diff
@@ -123,7 +131,15 @@
...
             android:layout_gravity="center_horizontal" />
-
+           
     </LinearLayout>
 
+     <include
+            android:id="@+id/brave_rewards_notification_permission_id"
+            layout="@layout/rewards_notification_permission_request"
+            android:visibility="gone"
+             />
```
Comment (deeppandya) [inline android/java/res/layout/brave_rewards_panel_notification_layout.xml:141]: nit : remove new line

File: android/java/res/layout/rewards_notification_permission_request.xml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        android:fontFamily="@font/poppins_400"
+        android:text="@string/close_text"
+        android:textColor="@color/rewards_panel_notification_secondary_text_color"
+        android:textSize="16sp"
+        android:layout_marginEnd="24dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="@+id/turnOnNotification"
+        app:layout_constraintTop_toBottomOf="@+id/turnOnNotification" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (deeppandya) [inline android/java/res/layout/rewards_notification_permission_request.xml:85]: nit : new line

File: android/java/res/layout/rewards_you_are_not_earning_dialog.xml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        android:clickable="true"
+        android:contentDescription="@null"
+        android:focusable="true"
+        app:srcCompat="@drawable/ic_baseline_close_24"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_chainStyle="packed"
+        app:tint="@color/onboarding_text_color" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (deeppandya) [inline android/java/res/layout/rewards_you_are_not_earning_dialog.xml:63]: nit : new line

**Review by deeppandya — APPROVED** [review]
LGTM

**Pull request merged**